tools/run-s3gw.sh

```diff
@@ -0,0 +1,209 @@
+#!/bin/bash
+#
+# run-s3gw.sh - wrapper to run a radosgw binary
+# Copyright 2022 SUSE, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+default_id="s3gw"
+default_debug_level="none"
+
+rgw_pid=
+
+usage() {
+  cat <<EOF
+usage: $0 [options... -- args]
+
+options:
+  --help                Shows this message.
+  --id ID               Specifies a custom instance ID (default: ${default_id}).
+  --cert FILE KEY       Specifies SSL certificate. Expects a .crt as first
+                        argument, and a .key as second argument.
+  --dns-name VALUE      For vhost style buckets, VALUE should be the DNS domain
+                        to be used.
+  --debug LEVEL         Runs with debug. Levels available: "high", "medium",
+                        "low", "none" (default: ${default_debug_level}).
+  --no-telemetry        Disable telemetry.
+  --telemetry-url URL   Specifies telemetry URL.
+  --with-status         Enables status frontend at 127.0.0.1:9999.
+
+
+args:
+  Any option supported by RADOS Gateway. For advanced use only.
+
+env variables:
+  S3GW_ID               Specifies a custom instance ID.
+  S3GW_DNS_NAME         Specifies a DNS domain to be used for vhost style
+                        buckets.
+  S3GW_DEBUG            Specifies the debug level to be used.
+  S3GW_CERT_FILE        Specifies the SSL certificate file.
+  S3GW_CERT_KEY         Specifies the SSL certificate key file.
+
+  Options always override environment variables.
+EOF
+}
+
+stop_rgw() {
+  kill -n SIGTERM ${rgw_pid}
+  exit 1
+}
+
+trap "stop_rgw" EXIT
```

Comment:
This needs to be moved down the bottom, just above when we run `radosgw` on line 199. Otherwise, if I try `docker run s3gw --help`, I get the help output, but I _also_ see "kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]" printed to STDERR. This is because `--help` triggers `exit 0` down on line 81, which hits the trap that calls `stop_rgw()`, which then tries to run `kill` with `$rgw_pid` as an empty string. Same with the various other exit cases. Moving `trap "stop_rgw" EXIT` down after those other `exit`s fixes this.